src/ophyd_async/core/_readable.py

```diff
@@ -123,30 +123,31 @@ def hints(self) -> Hints:
             # we want to combine them when they are Sequences, and ensure they are
             # identical when string values.
             for key, value in new_hint.hints.items():
+                # fail early for unkwon types
+                if not isinstance(value, str | Sequence):
+                    msg = (
+                        f"{new_hint.name}: Unknown type for value '{value}'"
+                        f" for key '{key}'"
+                    )
+                    raise TypeError(msg)
```

Comment:
Any reason to move this to the start of the if block? It means you're doing two `isinstance` checks for each of the happy paths, so is slightly less performant for the happy path which is the one we care about...

Reply:
I'll revert it back to the original order.
I personally like to fail early. But that's mostly personal preference.

Reply:
fixed: 9dfbb926dda864cdde1aeb3af76edde8816b9c92